src/components/TheSideDrawer.vue

```diff
@@ -185,8 +195,15 @@ export default {
       return displayDate(dateString);
     },
     toggleMic () {
-      this.isMicOn = !this.isMicOn
+      this.isMicOn = !this.isMicOn;
       this.$root.$emit('toggleMic', this.isMicOn);
+    },
+    async toggleHelpSignal () {
+      this.$root.$emit('toggle-help-signal', this.$route.params.room_id)
+      // let roomRef = db.doc(`classes/${this.classId}/blackboards/${this.$route.params.room_id}`)
```

Comment:
Right now we emit a global event and let ClassPageLiveBoard respond to the callback. However, I think it's better to just keep everything localized (by the principle of scope minimization) and update the Firestore doc directly in here. I'll make the change and push it as the next commit.

Reply:
Wait there must be a reason why you didn't do it since you have the exact code written out - let me investigate.

Reply:
Ah I see, room.status is not available.

Reply:
In Vuex, we store singleton data: user, class, etc. I added "room", so that it's now accessible by any component. I believe it's better to use global state (as Vuex is a well-established pattern) than global events, because root listeners violate the principle of scope minimization from 6.031, as now method's scope is global (one has to find both the root emitter and root listener). 

Now that the SideDrawer has access to the room, the method is now fully encapsulated in itself :)